### PR TITLE
Fixes #24680: Plugin backup status file format is incompatible between 8.0 and 8.1

### PR DESCRIPTION
**Pull request by Fdall**
https://issues.rudder.io/issues/24680

### CODE REVIEW

**Comment by Fdall** [comment]
PR updated with a new commit

**Comment by Fdall** [comment]
PR updated with a new commit

File: relay/sources/rudder-package/src/database.rs [hunk 1]
```diff
@@ -221,19 +221,64 @@ impl Database {
...
+            )
+        })?;
+        info!(
+            "Plugins statuses saved in {}",
+            backup_path.to_string_lossy()
+        );
         Ok(())
     }
 
+    fn read_plugin_status_line_from_backup(&self, line: &str, webapp: &mut Webapp) -> Result<()> {
```
Comment (amousset) [inline relay/sources/rudder-package/src/database.rs:237]:
```suggestion
    fn apply_plugin_status_line_from_backup(&self, line: &str, webapp: &mut Webapp) -> Result<()> {
```

it does more than reading

**Comment by Fdall** [comment]
PR updated with a new commit

File: relay/sources/rudder-package/src/lib.rs [hunk 2]
```diff
@@ -57,6 +57,7 @@ const SIGNATURE_KEYRING_PATH: &str = "/opt/rudder/etc/rudder-pkg/rudder_plugins_
 const RUDDER_VERSION_PATH: &str = "/opt/rudder/share/versions/rudder-server-version";
 const REPOSITORY_INDEX_PATH: &str = "/var/rudder/tmp/plugins/rpkg.index";
 const TMP_PLUGINS_FOLDER: &str = "/var/rudder/tmp/plugins";
+const PLUIN_STATUS_BACKUP_PATH: &str = "/tmp/rudder-plugins-upgrade";
```
Comment (amousset) [inline relay/sources/rudder-package/src/lib.rs:60]:
```suggestion
const PLUGIN_STATUS_BACKUP_PATH: &str = "/tmp/rudder-plugins-upgrade";
```

**Review by amousset — APPROVED** [review]

**Comment by Fdall** [comment]
OK, squash merging this PR

**Pull request merged**